nntrainer/layers/layer_node.cpp

```diff
@@ -420,10 +420,13 @@ void LayerNode::read(std::ifstream &file, bool opt_var) {
     << __func__ << " layer needs to be finalized first!";
   if (opt_var) {
     for (unsigned int i = 0; i < run_context->getNumWeights(); ++i) {
-      if (run_context->isGradientLastAccess(i)) {
+      if (run_context->isGradientLastAccess(i) && getTrainable()) {
         // @note read optimizer variables
-        for (unsigned int j = 0; j < run_context->getNumWeightOptVar(i); ++j) {
-          run_context->getWeightOptVar(i, j).read(file);
+        if (run_context->weightHasGradient(i)) {
```

Comment:
weightHasGradient() and getTrainable() have the same effect?

Reply:
it is not. for the batch normalization layer, it has 4 weights and two of them do not have gradient even if trainable is set true.

Reply:
thanks for the explanation.